.github/workflows/self-host-install.yml

```diff
@@ -15,27 +15,26 @@ on:
 jobs:
   touca-self-host-install:
     name: touca-self-host-install
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     timeout-minutes: 15
+    defaults:
+      run:
+        working-directory: ./api/e2e
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v4
         with:
-          python-version: 3.9
+          python-version: "3.10"
           cache: "pip"
+      - name: install dependencies
+        run: pip install .
```

Comment:
Kept `setup.py`. Now we'd use `pip install -e .[dev]` instead of `pip install -r requirements.txt`.